File: frontend-vibe/src/app/models/file-model.ts
import fileType from "./file-type";

export default class FileModel {
  id: number | null;
  name: string | null;
  fileType: fileType | null;
  downloadLink!: string | null;

  constructor(
    id: number | null = null,
    name: string | null = null,
    fileType: fileType | null = null,
    downloadLink: string | null = null
  ) {
    this.id = id,
      this.name = name,
      this.fileType = fileType,
      this.downloadLink = downloadLink
  }
}

File: VIBEBackend/AvatarService/Models/FileModel.cs
namespace AvatarService.Models
{
	public class FileModel
	{
		public long Id { get; set; }
		public string Name { get; set; }
		public string? FileType { get; set; }
		public string? DownloadLink { get; set; }

		public FileDto Convert()
		{
			var dto = new FileDto
			{
				Name = Name,
				FileType = FileType,
				DownloadLink = DownloadLink
			};

			return dto;
		}
	}
}